reduction/destreak.py

```diff
@@ -175,7 +181,21 @@ def destreak(frame, percentile=10, median_filter_size=256, overwrite=True, write
                                          )
 
     if use_background_map:
-        data = add_background_map(data, hdu, background_mapping=background_mapping)
+        proposal_id = hdu[0].header['PROGRAM'][1:5]
+        obsid = hdu[0].header['OBSERVTN'].strip()
+        if (proposal_id not in background_mapping or obsid not in background_mapping[proposal_id]):
+            print(f"WARNING: Either the proposal_id: {proposal_id} or obsid: {obsid} are not in background_mapping: {background_mapping}.")
+            hdu[('SCI', 1)].data = data
+            if write:
+                outname = frame.replace("_cal.fits", "_destreak.fits")
+                hdu.writeto(outname, overwrite=overwrite)
+                return outname
+            else:
+                return hdu
```

Comment:
All this code is duplicated.  Why not just skip this whole `if` block?

Reply:
i.e., change `if use _background_map:` to
```python
if use_background_map and not (proposal_id not in background_mapping or obsid not in background_mapping[proposal_id]):
   <do background map stuff>
```

Reply:
Good idea, should be fixed now